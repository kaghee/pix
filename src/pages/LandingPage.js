import React, { Component } from 'react';
import createRoom from '../chat';
import './Lobbies.scss';

export default class LandingPage extends Component {
  handleCreateRoom = () => {
    const {
      username,
      history,
    } = this.props;
    createRoom(username);
    history.push('/lobby');
  }

  render() {
    const { username } = this.props;

    return (
      <div className="start-wrapper">
        <div className="title">P I X I T</div>
        <div className="start-box">
          <div className="start-form">
            <input
              type="text"
              className="name"
              placeholder="Enter your name"
              maxLength="14"
              onChange={e => this.props.updateUsername(e.target.value)}
            />
            <div className="room-options">
              <div className="join-room">
                <input
                  type="text"
                  placeholder="Enter Room ID (optional)"
                  maxLength="20"
                  onChange={e => this.props.updateRoomName(e.target.value)}
                />
                <input className="btn join" type="button" value="Join Room" />
              </div>
              <input
                className="btn create"
                type="button"
                value="Create Private Room"
                onClick={() => this.handleCreateRoom(username)}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
