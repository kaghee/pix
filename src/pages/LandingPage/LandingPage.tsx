import { History } from 'history';
import React, { Component } from 'react';

import createRoom from '../../chat';

import '../Lobbies.css';

interface Props {
  username: string
  history: History
  updateUsername(username: string): void
  updateRoomName(roomName: string): void
}

export default class LandingPage extends Component<Props, {}> {
  handleCreateRoom = () => {
    const {
      username,
      history,
    } = this.props;
    createRoom(username);
    history.push('/lobby');
  }

  render() {
    return (
      <div className="start-wrapper">
        <div className="title">P I X I T</div>
        <div className="start-box">
          <div className="start-form">
            <input
              type="text"
              className="name"
              placeholder="Enter your name"
              maxLength={14}
              value={this.props.username}
              onChange={e => this.props.updateUsername(e.target.value)}
            />
            <div className="room-options">
              <div className="join-room">
                <input
                  type="text"
                  placeholder="Enter Room ID (optional)"
                  maxLength={20}
                  onChange={e => this.props.updateRoomName(e.target.value)}
                />
                <input className="btn join" type="button" value="Join Room" />
              </div>
              <input
                className="btn create"
                type="button"
                value="Create Private Room"
                onClick={() => this.handleCreateRoom()}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
