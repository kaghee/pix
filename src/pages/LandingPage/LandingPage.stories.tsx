import { action } from '@storybook/addon-actions';
import { text, withKnobs } from '@storybook/addon-knobs';
import { createMemoryHistory } from 'history';
import React from 'react';

import bg from '../../assets/images/blackhole.jpg';
import LandingPage from './';

export default {
  title: 'LandingPage',
  decorators: [withKnobs],
  parameters: {
    backgrounds: [
      { name: 'black', value: `black no-repeat center center/cover url(${bg})`, default: true },
    ],
  },
};

const landingPageActions = {
  updateUsername: action('updateUsername'),
  updateRoomName: action('updateRoomName'),
};

export const defaultRendering = () => (
  <LandingPage
    {...landingPageActions}
    username={text('Username', 'Kukor Ica')}
    history={createMemoryHistory()}
  />
);

defaultRendering.story = {
  name: 'Default rendering',
  notes: 'THE Default Rendering!',
};
