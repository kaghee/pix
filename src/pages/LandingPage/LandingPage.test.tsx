import { createMemoryHistory } from 'history';
import React from 'react';
import renderer from 'react-test-renderer';

import LandingPage from './';

test('Renders without error', () => {
  const component = renderer.create(
    <LandingPage
      username=""
      history={createMemoryHistory()}
      updateUsername={() => {}}
      updateRoomName={() => {}}
    />,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
