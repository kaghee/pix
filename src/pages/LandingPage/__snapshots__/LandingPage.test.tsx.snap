// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without error 1`] = `
<div
  className="start-wrapper"
>
  <div
    className="title"
  >
    P I X I T
  </div>
  <div
    className="start-box"
  >
    <div
      className="start-form"
    >
      <input
        className="name"
        maxLength={14}
        onChange={[Function]}
        placeholder="Enter your name"
        type="text"
      />
      <div
        className="room-options"
      >
        <div
          className="join-room"
        >
          <input
            maxLength={20}
            onChange={[Function]}
            placeholder="Enter Room ID (optional)"
            type="text"
          />
          <input
            className="btn join"
            type="button"
            value="Join Room"
          />
        </div>
        <input
          className="btn create"
          onClick={[Function]}
          type="button"
          value="Create Private Room"
        />
      </div>
    </div>
  </div>
</div>
`;
